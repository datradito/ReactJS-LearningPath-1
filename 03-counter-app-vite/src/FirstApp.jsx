import PropTypes from 'prop-types';

export const FirstApp = ( { title, subTitle, name } ) => {

    // console.log(props);

  return (
    <>
      <h1 data-testid="test-title"> { title } </h1>
      {/*<code>{ JSON.stringify( newMessage ) }</code>*/}
      <p>{ subTitle }</p>
      <p>{ subTitle }</p>
      <p>{ subTitle }</p>
      <p>{ name }</p>
    </>
  )
}


FirstApp.propTypes = {
    title: PropTypes.string.isRequired,
    subTitle: PropTypes.string,
}

FirstApp.defaultProps ={
    name: 'Jassira Ramos',
    subTitle: 'No hay subtitulo',
    //title: 'No hay titulo'
}
