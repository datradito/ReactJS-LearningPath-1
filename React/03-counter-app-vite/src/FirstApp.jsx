import PropTypes from 'prop-types';
// las importo para definirle el tipo a las properties, previamente descargo la dependencia (npm install prop-types);

export const FirstApp = ({ title, subTitle, name  }) => {
    //console.log(title);
    //trabajamos con el condicional, 
   {/* if (!title) {
        throw new Error ('el title no existe');
    } */}
  return (
    <>
    <h1 data-testid="test-title">{ title }</h1>
    <p>{ subTitle }</p>
    <p>{ subTitle }</p>
    <h2>{ name }</h2>
    </>
  )
}
//PROPTYPES DEFINIDOS en main: escribo los valores requeridos
FirstApp.PropTypes = {
title: PropTypes.string.isRequired,
subtitle: PropTypes.number.isRequired
}

{/**caso de utilizacion de numeros:  <p>{subtitle + 1}</p> y en el main  :subtitle = {1233}
 */}

 //tambien podremos definir las propiedades en un objeto: nota: se colocan al final

FirstApp.defaultProps = {
    name:'daiana herrera',
    subtitle: 'no hay sub titulo',
    // en seccion5 title: 'no hay titulo',
}