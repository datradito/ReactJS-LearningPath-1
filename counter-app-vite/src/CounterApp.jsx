import React from "react";
import { Fragment , useState} from "react";
import PropTypes from 'prop-types';




export const CounterApp = ({value}) => {

    const [counter, setCounter] =  useState(value);

    const  handleAdd = () =>  {
       
        setCounter(counter + 1 )

       /*  if(counter > 10) {
            alert('Legaste a 10!')
        } */
    
    }
    const  handleLess = () =>  {
       
        setCounter(counter - 1 )
    
    }
    const  handleReset= () =>  {
        setCounter(value)
    }

    return(
        <Fragment>
            <div className="container-counter">

                <h2>{counter}</h2>

                <button  aria-label="btn-suma" onClick={ handleAdd}>+</button>
                <button aria-label="btn-resta" onClick={ handleLess}>-</button>
                <button aria-label="btn-reset" onClick={ handleReset}>Reset</button>
            </div>
        </Fragment>
    )

}


CounterApp.propTypes = {

    value:  PropTypes.number.isRequired

}

CounterApp.defaultProps = {
    value:  0
}