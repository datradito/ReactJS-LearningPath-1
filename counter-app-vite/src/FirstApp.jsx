import React from "react";
import { Fragment } from "react";
import PropTypes  from 'prop-types'


const text =  "Welcome to my First App in React";

const person = {
    name: "Paloma",
    age: "21",
    f_color: "purple"
}

const sumar = (a,b) => {
    return a + b;
}

export const FirstApp = ({title , subtitle }) => {


    return(
       <Fragment>
           <header className="haeder">
                <div className="container-nav">

                    <nav className="nav">
                        <ul className="nav-list">
                            <li className="nav-list-item">Home</li>
                            <li className="nav-list-item">About me</li>
                            <li className="nav-list-item">Projects</li>
                        </ul>
                    </nav>

                </div>
                <div className="title" data-testid="test-title">
                    <h1>{title}</h1>
                    
                </div>
                <p>{subtitle}</p>
           </header>
            
           {/*  <h2>Hello World</h2>
            <p>This is my first app in react</p>

            <div class="container">

                <h3>Person</h3>
                <ul>
                    <li>{person.name}</li>
                    <li>{person.age}</li>
                    <li>{person.f_color}</li>
                </ul>

 
                <h3>Person</h3>
                <ul>
                    <li>{ JSON.stringify(person)}</li>
                   
                </ul>
            </div>

            <div>
                <p>La suma da como resultado: <span>{sumar(20,40)}</span> </p>
            </div> 

            <footer>
                <h6>By Paloma Herrera</h6>
            </footer>
        */}
       </Fragment>
    )
}



FirstApp.propTypes = {
    title: PropTypes.string.isRequired,
    subtitle: PropTypes.string.isRequired
}


FirstApp.defaultProps = {
    title: 'Hola, soy Paloma',
    subtitle:'Mi primera app react'

}