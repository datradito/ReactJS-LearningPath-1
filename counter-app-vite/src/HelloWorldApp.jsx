
/* function App() {

    const name = "Paloma";
    // document.createElement.... 
    return (<h1>Hello World, I am { name }</h1>)

}

export default App; */


export const HelloWorldApp = () => {
    return(
        <div>Hello World!!</div>
    )
}