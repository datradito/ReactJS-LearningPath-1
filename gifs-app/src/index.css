body{
  background: none;
  border: none;
  padding: 2em;
  background-color: black;
  color: white;
  font-family: 'Segoe UI', Tahoma, Geneva, Verdana, sans-serif;
  text-align: center;
  justify-content: center;
  
}
::placeholder { 
  color: rgb(244, 215, 255);
}
input{
  text-align: left;
  padding-right: 10px;
  outline: none;
}
input:hover, input:active{
  border: inherit;
  outline: none;
}
button, input{
  background: none;
  border: 0;
  color: inherit;
  background-color: rgb(119, 27, 204);
  color: rgb(255, 255, 255);
  font-size: 1.5em;
  padding: 1em;
}
button{
  cursor: pointer;
}
ol{
  list-style:decimal;
  text-align: justify;
}
.search{
  display: inline-block;
  width: 100%;
  /*background-color: rgb(79, 26, 114);*/
}
.GifGridItem{
  align-content: center;
  margin: 0.6em;
  border: 0.2em solid rgb(132, 0, 255);
}
.GifGridItem p {
  text-align: center;
}
.card-grid{
  align-content: center;
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
}
