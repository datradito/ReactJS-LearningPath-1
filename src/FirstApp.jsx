
import PropTypes from 'prop-types';

export const FirstApp = ({

    title,
    subTitle,
    name,
    valor
}) => {

    //if(!title) {throw new Error('el titulo no anda')};

    //console.log(title);
    return (
        <>
            <h1 data-testid="test-title"> {title} </h1>
            <p> {subTitle} </p>
            <p> {subTitle} </p>
            <p>{valor}</p>
            <p>{name}</p>

            <button onClick={function (event) { console.log(event) }}>
                +1
            </button>

        </>
    )
}

FirstApp.propTypes = {

    title: PropTypes.string.isRequired,
    subTitle: PropTypes.string.isRequired,
    valor: PropTypes.number.isRequired
};

FirstApp.defaultProps = {

    /* cancelado en clase 61 para testing 
     title: 'no hay titulo',*/
    subTitle: 'soy un subtitulo',
    name: "begoña",
    valor: 1
}

/*
export const FirstApp = ({ title }) => {

    console.log(props);


    return (
        <>
            <h1>
                Begoña
            </h1>
            {title}
            <p>soy un subtitulo</p>
        </>
    )
}

/*const getResult=(a,b)=>{
    return a+b;
}



const newMessage ={
    
    message:'hola mundo',
    title: 'begoña'
};

export const FirstApp = () => {

 
    
    return (
        //fragmento,agrupador elemento html
        //nodo padre
        <>
        <h1>{getResult(1,5)}</h1>{
        <code>{ JSON.stringify(newMessage)  }</code>

    <p>Soy un subtitulo</p>}
        </>
    )
};
*/