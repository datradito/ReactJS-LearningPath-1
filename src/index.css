html, body { 
    background-color: #212121;
    color: white;
    font-family: Arial, Helvetica, sans-serif;
    font-size: 1.3rem;
    padding:70%;

    }


    button{
        padding: 15px;
        font-size: 1.3rem;
        margin-right: 5px;
      
    }