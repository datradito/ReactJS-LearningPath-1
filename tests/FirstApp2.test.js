import { render, screen } from "@testing-library/react";
import { FirstApp } from "../src/FirstApp";

describe('Pruebas en <FirstApp/>', () => {

    const title = 'Hola, soy Begoña'

    test('debe de hacer match con el snapshot', () => {


        const { container } = render(<FirstApp title={title} />)
        expect(container).toMatchSnapshot();
    });


})

test('debe de mostrar el mensaje "hola soy Begoña"', () => {

    const title = 'Hola, soy Begoña'
    render(<FirstApp title={title} />)
    screen.debug();

    expect(screen.getAllByText(title)).toBeTruthy();
});


test('debe de mostrar el titulo en un h1', () => {

    const title = 'Hola, soy Begoña'
    render(<FirstApp title={title} />)
    screen.debug();

    expect(screen.getByText(title)).toBeTruthy();
});


const title = 'Hola, soy Begoña'
test('debe de mostrar el titulo en un h1', () => {
    render(<FirstApp title={title} />);
    expect(screen.getByRole('heading', { level: 1 }).innerHTML).toContain(title);

});

const subTitle = 'soy un subtitulo'
test('debe de mostrar el subtitulo enviado por props', () => {
    render(<FirstApp

        title={title}
        subTitle={subTitle}
    />);

    expect(screen.getAllByText(subTitle).length).toBe(2);


});