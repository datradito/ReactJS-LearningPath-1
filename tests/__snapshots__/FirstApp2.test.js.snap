// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <FirstApp/> debe de hacer match con el snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
     
    Hola, soy Begoña
     
  </h1>
  <p>
     
    soy un subtitulo
     
  </p>
  <p>
     
    soy un subtitulo
     
  </p>
  <p>
    1
  </p>
  <p>
    begoña
  </p>
  <button>
    +1
  </button>
</div>
`;
